Update _index.md based on TOC feedback (#1012)
falcosecurity/falco-website: content/en/about/why-falco.md

##### Compatibility with Container Orchestration Tools

> At its core, Falco is a kernel event monitoring and detection agent that captures events, such as syscalls, based on custom rules. Falco can enhance these events by integrating metadata from the container runtime and Kubernetes. This tight integration with various container orchestration tools enables the expansion of Falco's detection capabilities and scope. It can also detect and alert on new containers and workloads being deployed, ensuring that security visibility is comprehensive across your entire infrastructure. In addition, through Falco's native support for daemonset-like deployments, it can seamlessly integrate into your existing setup, whether you are using Kubernetes, Docker Swarm, or other container orchestration platforms.

Reviewer: > At its core, Falco is a kernel event monitoring and detection agent that captures events, such as syscalls, based on custom rules.

This sentence is not technically incorrect but can be misinterpreted. The word `captures` might give the impression that we are creating capturing files, or that we are outputting syscalls. The sentence also gives the impression that we are only capturing them based on custom rules. I would simplify the sentence to:

`At its core, Falco is a kernel event monitoring and detection agent.`

> across your entire infrastructure

This part can also be misinterpreted. I would say on the safe side and remove the word `entire`.

```suggestion
At its core, Falco is a kernel event monitoring and detection agent. Falco can enhance these events by integrating metadata from the container runtime and Kubernetes. This tight integration with various container orchestration tools enables the expansion of Falco's detection capabilities and scope. It can, for example, detect and alert on new containers and workloads being deployed, ensuring that security visibility is comprehensive across your infrastructure. In addition, through Falco's native support for daemonset-like deployments, it can seamlessly integrate into your existing setup, whether you are using Kubernetes, Docker Swarm, or other container orchestration platforms.
```

Reviewer 2: I second @pmusa :+1:  

In particular, I agree to remove the "captures" and "rules" since they can be misinterpreted and it's not relevant here since this paragraph mainly focuses on the compatibility with container orchestrators

Reviewer 3: This is great @pmusa would you like to update the main Falco readme as well since we have that sentence there currently? Just so we have everything cleaned up in one swing, I can quickly approve it.

##### Kernel-level instrumentation to capture system events

> Falco uses kernel-level instrumentation to capture system events by intercepting system calls and other kernel-level events in real-time. This approach allows Falco to detect security incidents more quickly and effectively, as it can analyze events before they are processed by the operating system or applications. Additionally, kernel-level instrumentation ensures that Falco is highly performant, as it can intercept and analyze events with minimal overhead.

Reviewer 4: Hi! I looked at this part in the context in which it's written. I believe this paragraph should be focused on the kernel component, like Aizhamal mentioned, and also avoid misconceptions related to interference with system calls but at the same time I don't think we should mention SIEMs or data lakes here as we're talking mostly about kernel instrumentation. So, here's my suggestion:

```
Falco uses kernel instrumentation to capture system events by intercepting system calls and other kernel-level signals. The Falco kernel components are designed to be fast and non intrusive, as they do not alter the system's behavior. By using this approach, Falco can collect rich information of what applications are doing nearly in real-time while minimizing overhead and preventing interference with regular workload behavior.
```

> Falco uses kernel-level instrumentation to capture system events by intercepting system calls and other kernel-level events in real-time. This approach allows Falco to detect security incidents more quickly and effectively, as it can analyze events before they are processed by the operating system or applications. Additionally, kernel-level instrumentation ensures that Falco is highly performant, as it can intercept and analyze events with minimal overhead.

Reviewer 4: I see that @leogr has responded at the same time as I did and couldn't read his response :sweat_smile: 
Anyways, I like the idea of using observing and monitoring as terms since this is what Falco does so I changed the suggestion accordingly.

> Falco uses kernel-level instrumentation to capture system events by intercepting system calls and other kernel-level events in real-time. This approach allows Falco to detect security incidents more quickly and effectively, as it can analyze events before they are processed by the operating system or applications. Additionally, kernel-level instrumentation ensures that Falco is highly performant, as it can intercept and analyze events with minimal overhead.

Reviewer 3: Love the rewrite and thanks for checking on the context here! @aijamalnk we should be good to go with this one! Cheers @LucaGuerra!

> Falco uses kernel-level instrumentation to capture system events by intercepting system calls and other kernel-level events in real-time. This approach allows Falco to detect abnormal behaviour and security threats quickly and effectively, as it can analyze events before they are processed by the operating system or applications. Additionally, kernel-level instrumentation ensures that Falco is highly performant, as it can intercept and analyze events with minimal overhead.

Reviewer 2: ```suggestion
Falco uses kernel-level instrumentation to monitor system events by observing system calls and other kernel-level events in real-time. This non-intrusive approach enables Falco to capture rich application activity signals and to detect and alert about abnormal behavior and security threats as soon as they happen. The events gathered can later be examined in SIEM or data lake systems. Additionally, Falco's developers are committed to maximizing the kernel instrumentation performance and ensuring events monitoring and processing come with minimal overhead.
```

- `capture` :point_right: `monitor` for the same reason explained by @pmusa in [this comment](https://github.com/falcosecurity/falco-website/pull/1012#discussion_r1239361221)
   - I also suggest modifying the title from `Kernel-level instrumentation to capture system events` :point_right: `Kernel-level instrumentation to monitor system events`
- `intercepting` :point_right: `observing` to make it clear that monitoring is passive and it doesn't interfere with system calls or other kernel activities and address the concern raised by @incertum in [this comment](https://github.com/falcosecurity/falco-website/pull/1012#discussion_r1236279628)
  - I've also added `non-intrusive approach` for the same reason.
 - Other rewrites inspired by @incertum in [this comment](https://github.com/falcosecurity/falco-website/pull/1012#discussion_r1236279628)

Reviewer 3: Yes @LucaGuerra rewrite with your inout is what we should go for https://github.com/falcosecurity/falco-website/pull/1012#discussion_r1240028542

@leogr similar to my comment above, now that we have clarified few things here can we re-audit the main Falco readme one more time and possibly adjust it with all the valuable information it should have?

Reviewer 2: @incertum agreed :+1: 

Anyway, I'd like to create a glossary or something like that to keep track of all definitions (i.e., monitoring vs capture). wdyt?

cc @vjjmiras

Reviewer 3: huh excellent idea love it!

> Falco uses kernel-level instrumentation to monitor system events by observing system calls and other kernel-level events in real-time. This non-intrusive approach enables Falco to capture rich application activity signals and to detect and alert about abnormal behavior and security threats as soon as they happen. The events gathered can later be examined in SIEM or data lake systems. Additionally, Falco's developers are committed to maximizing the kernel instrumentation performance and ensuring events monitoring and processing come with minimal overhead.

Author: ```suggestion
Falco uses kernel instrumentation to observe system events by monitoring system calls and other kernel-level signals. The Falco kernel components are designed to be fast and non intrusive, as they do not alter the system's behavior. By using this approach, Falco can collect rich information of what applications are doing nearly in real-time while minimizing overhead and preventing interference with regular workload behavior.
```
